packages/lit-html/src/lit-html.ts

```diff
@@ -920,9 +920,17 @@ class ChildPartImpl {
   _$setValue(value: unknown, directiveParent: DirectiveParent = this): void {
     value = resolveDirective(this, value, directiveParent);
     if (isPrimitive(value)) {
-      if (value === nothing) {
+      // Non-rendering child values. It's important that these do not render
+      // empty text nodes to avoid issues with preventing default <slot>
+      // fallback content.
+      if (
+        value === nothing ||
+        value === null ||
+        value === undefined ||
+        value === ''
+      ) {
         this._$clear();
-        this._$committedValue = nothing;
+        this._$committedValue = value;
```

Comment:
Sorta sucks that we don't have a dirty check for `nothing`; on a repeated render, it means we'll always do a `this._$startNode.nextSibling` DOM access for the part.

Worth adding the code? If so, leaving this as `nothing` and adding ` && this._ $committedValue !== nothing` to the conditions above would do it I think?

Reply:
Done.